### ISSUE TITLE
WAUTH-2 and WA 1.5 support checklist

### ISSUE THREAD

**Issue by matteocontrini** [hidespb/node-whatsapi#23]
√ done, – done but partially working, † still todo/to be checked

√ Online/offline status (719634957cdbb6d2cbeb741effb5202c553e0725)
√ Presence subscription (0c54edf21c71dff81cc0421984fa47a784fe623b)
– Request last seen (ea3a22948cfbe54809ed021e3b4167a867663a1d but see notes)
– Contacts sync (partially working 07cc03c1c27341c59bd78a934ef28cdb860216f9)
√ Message receipt (working but check `Node.prototype.shouldBeReplied`)
† Send composing/paused
† Send message
† Send location
† Send image
† Send video
† Send audio
† Send vCard
† Groups
† Set/get status (todo)
† Set profile picture
† Request profile picture
† Get server properties (todo)
† Get pricing (todo)
† Remove account (todo)
† Get/set privacy settings (todo)
† Send/get (?) client config (todo)

Feel free to edit when progress is made, at any time! :smile:

**Comment by tvandergeer** [comment]
Great idea to create this list!
Maybe we should add one extra state "documented" besides done, partialy working and todo/to be checked?

**Comment by matteocontrini** [comment]
You're right :+1: Added a new symbol (@)

**Comment by thiagupct** [comment]
Without ```{xmlns : 'http://jabber.org/protocol/chatstates'}``` in children ```sendComposing/sendPaused``` dosen't seems to work for me. Please check

**Comment by matteocontrini** [comment]
Are you sure? It's working fine by me. Are you calling `sendIsOnline` before?

**Comment by thiagupct** [comment]
Yes. Checked multiple times. I can't get it done without ```{xmlns : 'http://jabber.org/protocol/chatstates'} ```.So you check ``` sendComposing/sendPaused ``` with ```xmlns```. If this works, we can go with this.

**Comment by matteocontrini** [comment]
That `xmlns` attribute is not necessary since the tag itself is `chatstate`.
This code is working fine by me:

```
javascript
	wa.sendComposing('39xxx');
	setTimeout(function() {
		wa.sendPaused('39xxx');
	}, 2000);
```

Can you ( @thiagupct and @tvandergeer ) try it with the latest commit? It's working fine and both WhatsAPI and yowsup do that in this way.

**Comment by thiagupct** [comment]
It works now:-)

**Comment by matteocontrini** [comment]
I have started documenting v0.2 in the [wiki](https://github.com/hidespb/node-whatsapi/wiki). I hope I can get everything finished soon so we can merge that branch.

I will remove some events too, and replace them with callbacks of the functions they belong to. [Example](https://github.com/hidespb/node-whatsapi/wiki/Server-properties-request). The module should be easier to use this way.

**Comment by matteocontrini** [comment]
Most of the work should be completed. Functions have been rewritten using callbacks. Not everything is documented though.
I'd like to merge the WAUTH-2 branch in master anyway, and publish the update on npm, is that okay @tvandergeer ?

**Comment by matteocontrini** [comment]
Merged and published, only some documentation pages are missing now.
